packages/database/src/migrations/20210419225536-UpdatePSSSTotalSitesReportToUseSiteData-modifies-data.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import { generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const oldConfig = {
+  formula: 'PSSS_Sites_Reported',
+  aggregation: 'FINAL_EACH_WEEK',
+};
+
+const newConfig = {
+  formula: `firstExistingValue(PSSS_Sites_Reported, PSSS_Site_Weekly_Survey_Completed, PSSS_Site_Daily_Survey_Completed)`,
+  aggregation: {
+    PSSS_Sites_Reported: 'FINAL_EACH_WEEK',
+    PSSS_Site_Weekly_Survey_Completed: [
+      'FINAL_EACH_WEEK',
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'facility', aggregationEntityType: 'country' },
+      },
+    ],
+    PSSS_Site_Daily_Survey_Completed: [
+      'FINAL_EACH_WEEK',
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: { dataSourceEntityType: 'facility', aggregationEntityType: 'country' },
+      },
+    ],
+  },
+  defaultValues: {
+    PSSS_Sites_Reported: 'undefined',
+    PSSS_Site_Weekly_Survey_Completed: 'undefined',
+    PSSS_Site_Daily_Survey_Completed: 'undefined',
+  },
+};
+
+const PSSS_SITE_REPORTED_WEEKLY = {
+  code: 'PSSS_Site_Weekly_Survey_Completed',
+  builder: 'eventCheckConditions',
+  config: {
+    formula: 'true',
+    programCode: 'PSSS_WSR',
+  },
+};
+
+const PSSS_SITE_REPORTED_DAILY = {
+  code: 'PSSS_Site_Daily_Survey_Completed',
+  builder: 'eventCheckConditions',
+  config: {
+    formula: 'true',
+    programCode: 'PSSS_DSR',
+  },
+};
+
+const insertIndicator = async (db, indicator) => {
+  const { code } = indicator;
+
+  await insertObject(db, 'data_source', {
```

Comment:
Did you know that `meditrak-server` listens to indicator changes and updates data sources accordingly?? I added this logic a while back but I forgot! See the [Indicator model](https://github.com/beyondessential/tupaia/blob/dev/packages/database/src/modelClasses/Indicator.js#L34)

I guess it's fine in practice since the data source will be created first, and then the trigger will probably just fail because of a `code/type` constraint

Reply:
Oh wow! I didn't know! I still think explicitly creating them in migrations might be the way to go cos often when running locally you don't have meditrak-server up

Reply:
True!